src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This class holds ALL types of delegates that are coverted from ONE target method
+    /// Type parameters are generated for: <see cref="GetTypeArgumentsFromConversion(TypeSymbol, MethodSymbol)"/>
+    ///     1. All type parameters from the containing types of the target method
+    ///     2. All type parameters from the target method
+    ///     3. The delegate type
+    /// </summary>
+    /// <remarks>
+    /// This symbol is created in lowering phrase, and is collected and translated in emit phrase.
+    /// </remarks>
+    internal sealed class MethodGroupConversionCacheFrame : NamedTypeSymbol
```

Comment:
Yes, I originally defined the frame as `SynthesizedContainer`, that saved me a lot of work overriding abstract methods. However `SynthesizedContainer` was code to have a name when created, and uses type parameter symbols from anonymous type manager.

Well I guess I have to change that class.